apps/todo/app/page.tsx

```diff
@@ -1,144 +1,52 @@
+"use client";
 
-function Gradient({
-  conic,
-  className,
-  small,
-}: {
-  small?: boolean;
-  conic?: boolean;
-  className?: string;
-}): JSX.Element {
-  return (
-    <span
-      className={[
-        styles.gradient,
-        conic ? styles.glowConic : undefined,
-        small ? styles.gradientSmall : styles.gradientLarge,
-        className,
-      ]
-        .filter(Boolean)
-        .join(" ")}
-    />
-  );
-}
+import { FC, useState } from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
 
-const LINKS = [
-  {
-    title: "Docs",
-    href: "https://turbo.build/repo/docs",
-    description: "Find in-depth information about Turborepo features and API.",
-  },
-  {
-    title: "Learn",
-    href: "https://turbo.build/repo/docs/handbook",
-    description: "Learn more about monorepos with our handbook.",
-  },
-  {
-    title: "Templates",
-    href: "https://turbo.build/repo/docs/getting-started/from-example",
-    description: "Choose from over 15 examples and deploy with a single click.",
-  },
-  {
-    title: "Deploy",
-    href: "https://vercel.com/new",
-    description:
-      "Instantly deploy your Turborepo to a shareable URL with Vercel.",
-  },
-];
+type Input = {
+  item: string;
+};
 
-export default function Page(): JSX.Element {
-  return (
-    <main className={styles.main}>
-      <div className={styles.description}>
-        <p>
-          examples/basic&nbsp;
-          <Code className={styles.code}>web</Code>
-        </p>
-        <div>
-          <a
-            href="https://vercel.com?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo"
-            rel="noopener noreferrer"
-            target="_blank"
-          >
-            By{" "}
-            <Image
-              alt="Vercel Logo"
-              className={styles.vercelLogo}
-              height={24}
-              priority
-              src="/vercel.svg"
-              width={100}
-            />
-          </a>
-        </div>
-      </div>
+type Todo = {
+  id: number | null;
+  item: string;
+};
 
-      <Button appName="web" className={styles.button}>
-        Click me!
-      </Button>
+const TodoPage: FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const { register, handleSubmit, reset } = useForm<Input>();
 
-      <div className={styles.hero}>
-        <div className={styles.heroContent}>
-          <div className={styles.logos}>
-            <div className={styles.circles}>
-              <Image
-                alt=""
-                height={614}
-                src="circles.svg"
-                width={614}
-                style={{ pointerEvents: "none" }}
-              />
-            </div>
-            <div className={styles.logoGradientContainer}>
-              <Gradient className={styles.logoGradient} conic small />
-            </div>
+  const onSubmit: SubmitHandler<Input> = (data) => {
+    setTodoList(() => [...todoList, { id: todoList.length, item: data.item }]);
```

Comment:
`useState`のsetterはコールバック関数を引数にとれます。そのコールバック関数の引数（修正しているコードで言うと`prevTodoList`）は変更を加える前のステートです。ステートを更新する時には今のやりかたの更新ではバグるパターンがあります。TODOアプリケーションくらいの単純な構造では動きますが、1つのイベントハンドラの中で複数回ステートを更新するバグります。詳しくはドキュメントの[このページ](https://ja.react.dev/learn/state-as-a-snapshot)を見てみると良いかと思います。（わからなかったら聞いてね）
```suggestion
    setTodoList((prevTodoList) => [...prevTodoList, { id: prevTodoList.length, item: data.item }]);
```

Reply:
あってますかね？？
[9772ea1](https://github.com/peeek-develop-training/catchup-front/pull/4/commits/9772ea16ac92a03cadd573545f6c88ca831f5d32)

Reply:
あっています！
なんでこう修正したのか他の人が見ても分かるように記載して欲しいです！

Reply:
参考：https://ja.react.dev/reference/react/useState#updating-state-based-on-the-previous-state

`setTodoList(() => [...todoList, { id: todoList.length, item: data.item }]);`だと直前の`todoList`を更新できない
↓
`setTodoList((prev) => [...prev, { id: prev.length, item: data.item }]);`更新用関数：`prev`を呼び出す